pandas/tests/indexes/test_multi.py

```diff
@@ -1601,6 +1601,19 @@ def test_from_tuples(self):
         idx = MultiIndex.from_tuples(((1, 2), (3, 4)), names=['a', 'b'])
         self.assertEqual(len(idx), 2)
 
+    def test_from_tuples_variable_length(self):
+        # check that len(MultiIndex) == max(len(iterables))
+        T = ((1,), (2, 3), (4, 5, 6))
```

Comment:
yes not sure how to do this efficiently, this could potentially be hit in a lot of places, so pls check perf.